[Workers]: Add Momento integration (#10739)
cloudflare/cloudflare-docs: content/workers/databases/native-integrations/momento.md

---
pcx_content_type: configuration
title: Momento
---

# Momento

[Momento](https://gomomento.com/) is a truly serverless caching service. It automatically optimizes, scales, and manages your cache for you.

> {{<render file="_database_integrations_definition.md">}}

Reviewer: ```suggestion
{{<render file="_database_integrations_definition.md">}}
This integration allows you to connect to Momento from your Worker by getting Momento cache configuration and adding it as [secrets](https://1d791deb.cloudflare-docs-7ou.pages.dev/workers/configuration/environment-variables/) to your Worker.
```

## Momento Cache

To set up an integration with Momento Cache:

1. You need to have an existing Momento cache to connect to or create a new cache through the [Momento console](https://console.gomomento.com/).

> 2. Create a cache called `worker`. You'll need to make sure that you choose the same region when generating a Momento API key in the next step.

Author: No not really, I will change it

> 2. Create a cache called `user-profiles`. You'll need to make sure that you choose the same region when generating a Momento API key in the next step.

Reviewer 2: ```suggestion
2. Create a cache called `user-profiles`. You will need to make sure that you choose the same region when generating a Momento API key in the next step.
```

> 2. Create a cache called `user-profiles`. You'll need to make sure that you choose the same region when generating a Momento API key in the next step.

Reviewer 2: Why are they creating this cache if in step one they already have one?

Author: Step 2 assumes they don't have a cache and is showing the steps to create one. I'll phrase it better.

> 2. Create a cache called `user-profiles`. You will need to make sure that you choose the same region when generating a Momento API key in the next step.

Author: ```suggestion
2. To create a new cache, select create cache button and assign `user-profiles` as the cache name. You will need to make sure that you choose the same region when generating a Momento API key in the next step.
```

Author: @deadlypants1973 does this work?

3. Generate your [API key](https://docs.momentohq.com/develop/authentication/api-keys) to use within your application, select the cloud provider and region your application is in to get the best performance.

4. Add the Momento database integration to your Worker:

    1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
    2. In **Account Home**, select **Workers & Pages**.
    3. In **Overview**, select your Worker.
    4. Select **Integrations** > **Momento**. 
    5. Follow the setup flow, selecting the cache created in step 1.

> 5. The following example code show how to set an item in your cache, get it, and return it as a JSON object. The credentials needed to connect to Momento Cache have been automatically added as secrets to your Worker through the integration.

Reviewer 2: ```suggestion
5. The following example code show how to set an item in your cache, get it, and return it as a JSON object. The credentials needed to connect to Momento Cache have been automatically added as [secrets](/workers/configuration/secrets/) to your Worker through the integration.
```

Reviewer 2: Is the 'credentials needed' the API key in step 3? Because if so, let's say that. We didn't really connect how the API key is relevant to future steps in this doc.

Reviewer: Shouldn't be. Doesn't the integration automatically create and pull in an API key?

Author: it does! my bad I wrote this out before I had access to the integration. Now I followed the flow from the dash, and indeed the API key will be created by default so there's no need for the user to create one first. I've now added the steps in step 3.
